conda_lockfiles/create.py

```diff
@@ -76,11 +99,36 @@ def install_pypi_records(pypi_records: Iterable[str], prefix: str) -> CompletedP
         os.unlink(f.name)
 
 
-def install_conda_records(records: Iterable[PackageRecord], prefix: str) -> None:
+def lookup_conda_records(conda_specs: CondaSpecs) -> CondaRecords:
+    # normalize specs to a mapping
+    if not isinstance(conda_specs, Mapping):
+        conda_specs = {spec: {} for spec in conda_specs}
+    conda_specs = dict(conda_specs)
+
+    # populate package cache
+    pfe = ProgressiveFetchExtract(conda_specs.keys())
+    pfe.execute()
+
+    # lookup records in package cache
+    conda_records: list[PackageRecord] = []
+    for match_spec, overrides in conda_specs.items():
+        cache_record = next(PackageCacheData.query_all(match_spec), None)
```

Comment:
What if multiple packages match?

Reply:
We are coming from a lockfile, so we shouldn't have several entries for the same package name, so I guess we need to error out.